app/users/user.service.ts

```diff
@@ -102,119 +108,140 @@ export class UserService {
-    return this.cmsApiService.getUrl('/api/Users/?query=' + emailId + '&role=' + role, {})
+    return this.userStoreApiService.getUrl('/api/Users/ByEmail/' + emailId + '&role=' + role, {})
       .toPromise()
       .then(
-        (response: any) => User.extractPaginatedArrayData(response)
+      (response: any) => User.extractPaginatedArrayData(response)
       ).catch(
-        (error: any) => this.handleError(error)
+      (error: any) => this.handleError(error)
       );
   }
 
   /**
-   * Updates User Information
+   * Updates User Information. -- NEW USERSTORE API
    * @param user object with updated data
    * @param isCurrent updating the current user? Default value is false.
    */
-  public updateUser(user: User, isCurrent = false): Promise<any> {
-    return this.cmsApiService.putUrl('/api/User' + ( isCurrent ? '' : '?identity=' + user.email ), JSON.stringify(user), {})
+  public updateUser(user: User): Promise<any> {
+    return this.userStoreApiService.putUrl('/api/Users/' + user.id, JSON.stringify(user), {})
       .toPromise()
+      .then(
+      (response: Response) => {
+        let localUser = this.userCache.getValue();
+        let userToUpdate = localUser.find(item => item.id === user.id);
+        for (let prop in userToUpdate) {
+          if (userToUpdate.hasOwnProperty(prop)) {
+            userToUpdate[prop] = user[prop];
+          }
+        }
+        this.userCache.next(localUser);
+        return response;
+      }
+      )
       .catch(
-        (error: any) => this.handleError(error)
+      (error: any) => this.handleError(error)
       );
   }
 
-  public getPicture(identifier: string, useCurrent = false): Promise<any> {
-    return this.cmsApiService.getUrl('/api/User/Picture' + (useCurrent ? '' : '?identity=' + identifier), {})
+  public getPicture(id: string, useCurrent = false): Promise<any> {
+    let headers = new Headers();
+    headers.append('Accept', 'application/json');
+    let options = new RequestOptions({ headers: headers, responseType: ResponseContentType.ArrayBuffer });
+    return this.userStoreApiService.getUrl('/api/Users/' + id + '/Photo', options)
       .toPromise()
+      .then(
+      (response => UserService.extractContent(response, true))
+      )
       .catch(
-        (error: any) => this.handleError(error)
+      (error: any) => this.handleError(error)
       );
   }
 
-  public uploadPicture(fileToUpload: any, identifier: string) {
+  public uploadPicture(fileToUpload: any, id: string) {
     let data = new FormData();
     data.append('file', fileToUpload);
-    return this.cmsApiService.putUrlWithFormData('/api/User/Picture?identity=' + identifier, data)
-       .toPromise()
-       .catch(
-         (error: any) => this.handleError(error)
-       );
+    return this.userStoreApiService.putUrlWithFormData('/api/Users/' + id + '/Photo', data)
+      .toPromise()
+      .then(
+      (response: any) => (response.status === 200)
```

Comment:
Is this `then` even necessary?

Reply:
Yeah..i tried to remove it but it was giving me error...i guess we can keep it..